### PR TITLE
Add optional reason to @PendingFeature

### PR DESCRIPTION
**Pull request by leonard84**

### CODE REVIEW

File: spock-specs/src/test/groovy/org/spockframework/smoke/extension/PendingFeatureExtensionSpec.groovy [hunk 1]
```diff
@@ -23,6 +23,26 @@ class Foo extends Specification {
...
+  }
+}
+    """)
+
+    then:
+    notThrown(AssertionError)
+    result.runCount == 1
+    result.failureCount == 0
+    result.ignoreCount == 0
+    // unfortunately we can't access the skip reason here, but at least the branch is coverd
```
Comment (marcphilipp) [inline spock-specs/src/test/groovy/org/spockframework/smoke/extension/PendingFeatureExtensionSpec.groovy]: Nit: typo in last word

File: spock-core/src/main/java/org/spockframework/runtime/extension/builtin/PendingFeatureBaseInterceptor.java [hunk 2]
```diff
@@ -25,6 +27,7 @@ protected AssertionError featurePassedUnexpectedly() {
   }
 
   protected AssumptionViolatedException assumptionViolation() {
-    return new AssumptionViolatedException("Feature not yet implemented correctly.");
+    return new AssumptionViolatedException("Feature not yet implemented correctly."
+      + ("".equals(reason) ? "" : "Reason: "+reason));
```
Comment (marcphilipp) [inline spock-core/src/main/java/org/spockframework/runtime/extension/builtin/PendingFeatureBaseInterceptor.java]: If I'm not mistaken there's no space after the period now. It should be `" Reason: "`. Moreover spaces around the `+` would be more readable IMHO.

File: docs/release_notes.adoc [hunk 3]
```diff
@@ -6,6 +6,8 @@ include::include.adoc[]
 
 === What's New In This release
 
+* Improve `@PendingFeature` now has an optional `reason` attribute
```
Comment (marcphilipp) [inline docs/release_notes.adoc]:
```
* Improve `@PendingFeature` to now have an optional `reason` attribute`
```
?

**Comment by leonard84** [comment]
PTAL

**Pull request merged**